es/13_3_Potenciando_Bitcoin_con_Scripts.md

```diff
@@ -0,0 +1,317 @@
+
+```
+Script: <sigAlice> <revokeCode> OP_HASH160 <revokeHash> OP_EQUAL IF <pubKeyBob> ELSE <+24Hours> OP_CHECKSEQUENCEVERIFY OP_DROP <pubKeyAlice> ENDIF OP_CHECKSIG
+Stack: [ ]
+
+Script: OP_HASH160 <revokeHash> OP_EQUAL IF <pubKeyBob> ELSE <+24Hours> OP_CHECKSEQUENCEVERIFY OP_DROP <pubKeyAlice> ENDIF OP_CHECKSIG
+Stack: [ <sigAlice> <revokeCode> ]
+
+Script: <revokeHash> OP_EQUAL IF <pubKeyBob> ELSE <+24Hours> OP_CHECKSEQUENCEVERIFY OP_DROP <pubKeyAlice> ENDIF OP_CHECKSIG
+Running: <revokeCode> OP_HASH160 
+Stack: [ <sigAlice> <revokeHash> ]
+
+Script: OP_EQUAL IF <pubKeyBob> ELSE <+24Hours> OP_CHECKSEQUENCEVERIFY OP_DROP <pubKeyAlice> ENDIF OP_CHECKSIG
+Stack: [ <sigAlice> <revokeHash> <revokeHash> ]
+
+Script: IF <pubKeyBob> ELSE <+24Hours> OP_CHECKSEQUENCEVERIFY OP_DROP <pubKeyAlice> ENDIF OP_CHECKSIG
+Running: <revokeHash> <revokeHash> OP_EQUAL
+Stack: [ <sigAlice> True ]
+```
+
+El `OP_EQUAL` alimenta la declaración` IF`. Debido a que Alice usa el `revokeCode`, ingresa a la sucursal que le permite canjear los fondos inmediatamente, colapsando el resto del script a` <pubKeyBob> `(dentro del condicional) y`OP_CHECKSIG` (después).
+
+```
+Script: <pubKeyBob> OP_CHECKSIG
+Running: True IF
+Stack: [ <sigAlice> ]
+```
+Maldiciones! ¡Solo Bob puede firmar inmediatamente usando el `redeemCode`!
+
+```
+Script:  OP_CHECKSIG
+Stack: [ <sigAlice> <pubKeyBob> ]
+
+Script:
+Running: <sigAlice> <pubKeyBob> OP_CHECKSIG
+Stack: [ False ]
+```
+
+#### Ejecuta el script de bloqueo para engañar a Alice, sin código de revocación
+
+Entonces, ¿qué pasa si Alice intenta usar su propia firma, sin el `revokeCode`? Ella usa un script de desbloqueo de `<sigAlice> <notRevokeCode>`.
+
+```
+Script: <sigAlice> 0 OP_HASH160 <revokeHash> OP_EQUAL IF <pubKeyBob> ELSE <+24Hours> OP_CHECKSEQUENCEVERIFY OP_DROP <pubKeyAlice> ENDIF OP_CHECKSIG
+Stack: [ ]
+
+Script: OP_HASH160 <revokeHash> OP_EQUAL IF <pubKeyBob> ELSE <+24Hours> OP_CHECKSEQUENCEVERIFY OP_DROP <pubKeyAlice> ENDIF OP_CHECKSIG
+Stack: [ <sigAlice> 0 ]
+
+Script: <revokeHash> OP_EQUAL IF <pubKeyBob> ELSE <+24Hours> OP_CHECKSEQUENCEVERIFY OP_DROP <pubKeyAlice> ENDIF OP_CHECKSIG
+Running: 0 OP_HASH160 
+Stack: [ <sigAlice> <0Hash> ]
+
+Script: OP_EQUAL IF <pubKeyBob> ELSE <+24Hours> OP_CHECKSEQUENCEVERIFY OP_DROP <pubKeyAlice> ENDIF OP_CHECKSIG
+Stack: [ <sigAlice> <0Hash> <revokeHash> ]
+
+Script: IF <pubKeyBob> ELSE <+24Hours> OP_CHECKSEQUENCEVERIFY OP_DROP <pubKeyAlice> ENDIF OP_CHECKSIG
+Running: <0Hash> <revokeHash> OP_EQUAL
+Stack: [ <sigAlice> False ]
+```
+
+Ahora colapsamos hasta la declaración `ELSE` y lo que viene después del condicional:
+
+```
+Script: <+24Hours> OP_CHECKSEQUENCEVERIFY OP_DROP <pubKeyAlice> OP_CHECKSIG
+Running: False IF
+Stack: [ <sigAlice> ]
+
+Script: OP_CHECKSEQUENCEVERIFY OP_DROP <pubKeyAlice> OP_CHECKSIG
+Stack: [ <sigAlice> <+24Hours> ]
+```
+
+¡Y luego Alice está frustrada nuevamente porque no han pasado 24 horas!
```

Comment:
- `¡Y luego Alice es frustrada nuevamente porque no han pasado 24 horas!`

Reply:
Yes, like: `El suceso paso de la siguiente manera: entran a robar, bla bla bla, pero el asalto es frustrado por la policia`. I understand the meaning of the sentence as: `The mischievous attempt of Alice to steal money from Bob was frustrated` but with the verb in present instead of past simple (passive tense).

Reply:
It could be "se siente frustrada".   But "es frustrada" it's wrong